### DISCUSSION TITLE
Can I fix buildId when output export?

### DISCUSSION THREAD

**Discussion by zao95** [vercel/next.js#74414] [Help]
### Summary

I'm using the next 14.2.13 version, and it's an SSG build environment through `output: 'export'` setting.

There is a limit to the number of files when deploy on my company system,
I want to fix buildId by modifying the next.config.js file as below.

```js
generateBuildId: async () => {
    return "fixedBuildId"
  },
```

In this process, I am considering settings because I am worried about cache processing.

I plan to set the `Cache-Control: no-cache, no-store` header to respond to the two files in case `/out/static/fixedBuildId/_buildManifest.js` and `/out/static/fixedBuildId/_ssgManifest.js` change during the build path.

Other than this, are there any additional side effects from fixing buildId?

Additionally, it seems that the list of chunks for pages written in the past `_buildManifest.js` is written at `{path}.txt` in each path, but doesn't the txt file also need a `Cache-Control` header?

**Comment by icyJoseph** [comment]
I remember someone asked this way before, a few years back perhaps, and they reported back that it had caused some issues. Let me find it for you.

**Reply by icyJoseph** [reply]
https://github.com/vercel/next.js/discussions/53346#discussioncomment-6986761

While they don't give too much insight, my guess is that, upon version changes, the module exports and such might diverge, and/or the loaded modules won't be as expected by the loader module, leading to JavaScript exceptions.

**Reply by icyJoseph** [reply]
Maybe use the last GIT Head as build id? something like this, https://gist.github.com/giautm/742aef6fe3c1bede9e4a92e71715a1c2

**Reply by zao95** [reply]
There are more than 800 pages in the project, but the maximum file changes when deployed on the company system are limited to 200.
The buildId changes even when using githead,
I understand that each page has a change based on buildId, so I don't think it will fit with the system.

I'll check the attached thread.
Thank you for your answer.

**Reply by icyJoseph** [reply]
Aha, I think you want to fix the webpack chunk ids rather...

**Reply by icyJoseph** [reply]
Here ya go: https://github.com/vercel/next.js/discussions/65856#discussioncomment-9749287

It looks like, potentially, a bug is at play here as well... Read the whole thread I'm linking, not just the workaround.

**Reply by zao95** [reply]
The thread you posted confirms that when you build generateBuildId in two or more separate instances, the results may be different, is it correct that I understand?
Fortunately, in my environment, the build results are copied, so I don't think it will be applied to the error case above.

**Reply by zao95** [reply]
During testing, the hash values used for page and chunk file names change only when the code changes, in the case of the pages router,
However, when you configure it as an app router, all files change every time you build it
Oh, my..
